tools/benchmark/src/mocha/customOutputRunner.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Test } from "mocha";
+
+import type { BenchmarkDescription, MochaExclusiveOptions } from "../Configuration";
+import { timer } from "../timer";
+
+/**
+ * Options to configure a benchmark that reports custom measurements.
+ *
+ * @public
+ */
+export interface CustomBenchmarkOptions extends BenchmarkDescription, MochaExclusiveOptions {
+	/**
+	 * Title about benchmark option
+	 */
+	title: string;
+
+	/**
+	 * Runs the benchmark.
+	 */
+	run: (reporter: IMeasurementReporter) => void | Promise<unknown>;
```

Comment:
Should this return `void | Promise<void>`? It seems odd to me that the synchronous version of this doesn't return anything, but the async version returns something.

Reply:
Maybe the return type should be only one of them? 

This change was part of [21555](https://github.com/microsoft/FluidFramework/pull/21555). 

Looping in @chentong7 for more information.

Reply:
Hmm... runBenchmarkSync() does return the benchmark result object `BenchmarkData` for runtime reporter to log the result. This run() has reporter included inside, so I think it might not necessary to have a return object for the custom data.